AirCasting/SessionViews/ChartView.swift

```diff
@@ -52,26 +52,30 @@ class UI_PollutionChart: UIView {
 }
 
 struct ChartView: UIViewRepresentable {
-    
-    @ObservedObject var stream: MeasurementStreamEntity
-    var thresholds: [SensorThreshold]
+    private let stream: MeasurementStreamEntity
+    private let thresholds: [SensorThreshold]
 
+    @StateObject var chartCreator: ChartViewModel
+    
     typealias UIViewType = UI_PollutionChart
     
+    init(stream: MeasurementStreamEntity, thresholds: [SensorThreshold]) {
+        self.stream = stream
+        self.thresholds = thresholds
+        self._chartCreator = .init(wrappedValue: ChartViewModel(stream: stream))
+    }
+    
     func makeUIView(context: Context) -> UI_PollutionChart {
         UI_PollutionChart()
     }
     
     func updateUIView(_ uiView: UI_PollutionChart, context: Context) {
-        let chartCreator = ChartEntriesCreator(stream: stream)
-        var entries = chartCreator.generateEntries()
+        guard !chartCreator.entries.isEmpty else { return }
         
-        if entries.isEmpty {
-            return()
-        } else {
-            entries.sort { (e1, e2) -> Bool in
-                e1.x < e2.x
-            }
+        var entries = chartCreator.entries
+        
+        entries.sort { (e1, e2) -> Bool in
+            e1.x < e2.x
```

Comment:
This is a pretty common thing to do (sort over some property of an object), how about we make it nice and clean?

I've come up with a simple set of extensions that will allow us to do just:
```Swift
entries.sort(by: \.x, areInIncreasingOrder: <)
```
which I think is pretty neat. Just use my code:
```Swift
extension Sequence {
    func sorted<T: Comparable>(by keyPath: KeyPath<Element, T>, areInIncreasingOrder: (T, T) throws -> Bool) rethrows -> [Element] {
        try sorted {
            try areInIncreasingOrder($0[keyPath: keyPath], $1[keyPath: keyPath])
        }
    }
}

extension Array {
    mutating func sort<T: Comparable>(by keyPath: KeyPath<Element, T>, areInIncreasingOrder: (T, T) throws -> Bool) rethrows {
        self = try sorted(by: keyPath, areInIncreasingOrder: areInIncreasingOrder)
    }
}
```

Reply:
I personally think it makes it a bit more confusing, so I'd leave it as is.

Reply:
Consider creating something on top of this general extensions like:
```Swift
entries.sort(by: \.x, ascending: true/false)
```

or even with default value it can get to this point:
```Swift
entries.sort(by: \.x)
```
I think it's much better

Also - leaving readability and code clarity aside - note that by using keypaths you'd also eliminate a whole class of typo errors, for example this one:
```Swift
cars.sort {
   $0.engineCapacity < $0.engineCylinders
}
```
won't be possible.

But I'm not forcing it that hard - if you feel it's not right - just leave it :)

Reply:
I tried implementing it, but it didn't work. However, this: `areInIncreasingOrder: <` is a bit confusing to me, but I guess when I want it ti be in the reversed order (descreasing) I should do it like: `areInIncreasingOrder: >`? Anyway, I pushed it to a different [branch](https://github.com/HabitatMap/AirCastingiOS/tree/new-sorting) so you can check what I did wrong. For now, I'm reverting it to the previous version.